sdk/core/azure-core/src/main/java/com/azure/core/http/HttpPipelineBuilder.java

```diff
@@ -86,24 +88,32 @@ public HttpPipeline build() {
             client = HttpClient.createDefault();
         }
 
-        configureTracing(policies);
+        configureTracing(policies, clientOptions);
 
         return new HttpPipeline(client, policies, tracer);
     }
 
-    private void configureTracing(List<HttpPipelinePolicy> policies) {
+    private void configureTracing(List<HttpPipelinePolicy> policies, ClientOptions clientOptions) {
         if (tracer == null) {
             TracingOptions tracingOptions = clientOptions == null ? null : clientOptions.getTracingOptions();
             tracer = TracerProvider.getDefaultProvider().createTracer("azure-core", null, null, tracingOptions);
         }
-
         for (HttpPipelinePolicy policy : policies) {
             if (policy instanceof InstrumentationPolicy) {
-                ((InstrumentationPolicy) policy).initialize(tracer);
+                UrlSanitizer sanitizer = new UrlSanitizer(getAllowedQueryParams(clientOptions));
+                ((InstrumentationPolicy) policy).initialize(tracer, sanitizer);
             }
         }
     }
 
+    private static Set<String> getAllowedQueryParams(ClientOptions options) {
```

Comment:
nit: consider naming this method `getAllowedTracingQueryParams()` to scope this to query params set in `TracingOptions`.